### ISSUE TITLE
forbid-prop-types warning shown on Yup validation

### ISSUE THREAD

**Issue by MerlinMason** [jsx-eslint/eslint-plugin-react#2722]
Hey there, a project I work on uses both prop-types and [Yup](https://github.com/jquense/yup) which have a similar syntax.

The following where I'm declaring a Yup validation object:
```js
const formValidation = Yup.object().shape({
    name: Yup.string(),
    customer_ids: Yup.array()
});
```

Gives me a warning on the line `customer_ids: Yup.array()` like so:
```
Prop type `array` is forbidden eslint(react/forbid-prop-types)
```

I believe this is a bug as this is not a prop type declaration.

**Comment by ljharb** [comment]
This should only warn when object/array/etc is chained off of `React.PropTypes` or an import/require of the `prop-types` package.

**Comment by codecog** [comment]
Working on this

**Comment by MParolariDoveconviene** [comment]
I have the same problem with this declaration

```
const validation = yup.object().shape({
  address: yup.object({
    city: yup.string(),
    zip: yup.string(),
  })
})
```

I found the problem started form version  7.20.1, in particular with this [commit](https://github.com/yannickcr/eslint-plugin-react/commit/997a99f389a7df9673de890d71a17e2b6838f72b). From what I understand, it checks recursively for the propTypes definition to be correct. I tried to fix it adding  to`CallExpression` in `lib/rules/forbid-prop-types.js` a check for the parent to be a propTypes declaration using `propsUtil.isPropTypesDeclaration` but it works only for the direct children of PropTypes and breaks the recursive behavior. 
Maybe I should block the recursive calls in another point?

**Comment by Parro** [comment]
I have the same problem with this declaration
      
```
const validation = yup.object().shape({
  address: yup.object({
    city: yup.string(),
    zip: yup.string(),
  }) 
})
```

 I found the problem started form version  7.20.1, in particular with this [commit](https://github.com/yannickcr/eslint-plugin-react/commit/997a99f389a7df9673de890d71a17e2b6838f72b). From what I understand, it checks recursively for the propTypes definition to be correct. I tried to fix it adding  to`CallExpression` in `lib/rules/forbid-prop-types.js` a check for the parent to be a propTypes declaration using `propsUtil.isPropTypesDeclaration` but it works only for the direct children of PropTypes and breaks the recursive behavior.

Maybe I should block the recursive calls in another point?

**Comment by rafaelnogueira1** [comment]
I don't know if anyone is still having this problem but i update some of my packages today and got a similar error when running my lint, `Prop type `array` is forbidden`.

I was able to fix by changing this:
```
Yup.array(
      Yup.object().shape({ 
            value: Yup.number()
      })
)
```

For this:
```
Yup.array().of(
      Yup.object().shape({ 
            value: Yup.number()
      })
)
```

**Comment by ljharb** [comment]
Go for it!
